src/read_resume_data.cpp

```diff
@@ -165,13 +166,14 @@ namespace libtorrent
 		if (file_priority)
 		{
 			int const num_files = file_priority.list_size();
-			ret.file_priorities.resize(num_files, 4);
+			ret.file_priorities.resize(aux::numeric_cast<std::size_t>(num_files), 4);
 			for (int i = 0; i < num_files; ++i)
 			{
-				ret.file_priorities[i] = std::uint8_t(
+				std::size_t const idx = std::size_t(i);
+				ret.file_priorities[idx] = aux::numeric_cast<std::uint8_t>(
 					file_priority.list_int_value_at(i, 1));
```

Comment:
I also wonder if this value should be clamped with run-time checks, rather than asserted